tests/Sharpliner.Serialization.Tests/MockPipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Sharpliner.Model.AzureDevOps;
+
+namespace Sharpliner.Serialization.Tests
+{
+    internal class MockPipeline : AzureDevOpsPipelineDefinition
+    {
+        public override string TargetFile => "azure-pipelines.yml";
+
+        public override TargetPathType TargetPathType => TargetPathType.RelativeToGitRoot;
+
+        public override Pipeline Pipeline => new()
+        {
+            Name = "$(Date:yyyMMdd).$(Rev:rr)",
+
+            Variables =
+            {
+                new Variable("Configuration", "Release"), // We can create the objects and then resue them for definition too
+                Variable("Configuration", "Release"),     // Or we have this more YAML-like definition
+                Group("PR keyvault variables"),
+
+                If.Equal("variables['Build.Reason']", "PullRequest")
+                    .Variable("TargetBranch", "$(System.PullRequest.SourceBranch)")
+                    .Variable("IsPr", true),
+
+                If.And(Equal("variables['Build.SourceBranch']", "refs/heads/production"), NotEqual("Configuration", "Debug"))
```

Comment:
Can we have:
```suggestion
                If.And(Equal(Variable(Build.SourceBranch"), "refs/heads/production"), NotEqual("Configuration", "Debug"))
```
We should be able to infer that is a variable and later do write the 'variables['Build.SourceBranch'] inside the if model.

Reply:
We can even have 
```
Equal(variables["Build.SourceBranch"], "refs/heads/production")
```

What do you say?

Reply:
@mandel-macaque I updated the `variables[]` usage - now it's almost 100% like YAML :)